extensions-contrib/momentsketch/src/main/java/org/apache/druid/query/aggregation/momentsketch/aggregator/MomentSketchBuildBufferAggregator.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.momentsketch.aggregator;
+
+import org.apache.druid.query.aggregation.BufferAggregator;
+import org.apache.druid.query.aggregation.momentsketch.MomentSketchWrapper;
+import org.apache.druid.segment.ColumnValueSelector;
+
+import java.nio.ByteBuffer;
+
+public class MomentSketchBuildBufferAggregator implements BufferAggregator
+{
+  private final ColumnValueSelector<Double> selector;
+  private final int k;
+  private final boolean compress;
+
+  public MomentSketchBuildBufferAggregator(
+      final ColumnValueSelector<Double> valueSelector,
+      final int k,
+      final boolean compress
+  )
+  {
+    this.selector = valueSelector;
+    this.k = k;
+    this.compress = compress;
+  }
+
+  @Override
+  public synchronized void init(final ByteBuffer buffer, final int position)
+  {
+    ByteBuffer mutationBuffer = buffer.duplicate();
+    mutationBuffer.position(position);
+
+    MomentSketchWrapper emptyStruct = new MomentSketchWrapper(k);
+    emptyStruct.setCompressed(compress);
+    emptyStruct.toBytes(mutationBuffer);
+  }
+
+  @Override
+  public synchronized void aggregate(final ByteBuffer buffer, final int position)
```

Comment:
Should use finer, read-write concurrency; see `HllSketchBuildBufferAggregator` for example

Reply:
I was told here https://github.com/apache/incubator-druid/pull/6381 that synchronization is not needed in buffer aggregators.

Reply:
@leventov do you suggestions on how to proceed here?

Reply:
the buffer aggregator doesn't need synchronization, but the non-buffer aggregator does since those are used in realtime ingestion tasks which can be queried as they're ingesting

Reply:
Some buffer aggregators do need synchronization for `OffheapIncrementalIndex`. See #3956.

Reply:
OK I will keep both the buffer and non-buffer aggregators synchronized then.